lib/base-compiler.ts

```diff
@@ -1695,11 +1696,13 @@ export class BaseCompiler implements ICompiler {
         };
     }
 
-    async getOrBuildExecutable(key) {
+    async getOrBuildExecutable(key, bypassCache?: BypassCacheControl) {
         const dirPath = await this.newTempDir();
 
-        const buildResults = await this.loadPackageWithExecutable(key, dirPath);
-        if (buildResults) return buildResults;
+        if (!bypassCache || !(bypassCache & BypassCacheControl.Compilation)) {
```

Comment:
can you write this with && ? using a bitwise operator for the first time in the codebase seems confusing

Reply:
`bypassCache` is a flag enum value, this check could be abstracted to a function but I don't think it'd be entirely warranted to do so.

Reply:
Well, thinking about it more I can't imagine it ever making sense to recompile but not also re-execute, so maybe everything can be rephrased not in terms of a flag enum.